common/metrics.py

```
@@ -0,0 +1,223 @@
+from typing import Any, ContextManager, Dict, TypeVar, Union
```

Ruff (0.12.2)

`typing.ContextManager` is deprecated, use `contextlib.AbstractContextManager` instead

(UP035)